core/src/autogluon/core/scheduler/seq_scheduler.py

```diff
@@ -86,15 +87,15 @@ class LocalSequentialScheduler(object):
         Note: The type of resource must be int.
     """
 
-    def __init__(self, train_fn, search_space, util_args=None, searcher='auto', **kwargs):
+    def __init__(self, train_fn, search_space, util_args=None, searcher='auto', reward_attr='reward', resource=None, **kwargs):
         self.train_fn = train_fn
         self.training_history = None
         self.config_history = None
-        self._reward_attr = kwargs['reward_attr']
+        self._reward_attr = reward_attr
         self.time_attr = kwargs.get('time_attr', None)
-        self.resource = kwargs['resource']
+        self.resource = resource
         self.max_reward = kwargs.get('max_reward', None)
-        self.searcher: BaseSearcher = self.get_searcher_(searcher, train_fn, search_space=search_space, **kwargs)
+        self.searcher: LocalSearcher = self.get_searcher_(searcher, train_fn, search_space=search_space, **kwargs)
```

Comment:
Why replace `BaseSearcher` with less abstract version of this class? I'm thinking about a case when DistributedSearcher is added and LocalSeracher is too specific here.

Reply:
Unclear what BaseSearcher would look like atm. Can revisit after finish removing outdated code.

Reply:
LocalSearcher does not depend on BaseSearcher, I'm deleting BaseSearcher in this PR.